components/dashboards/org.wso2.carbon.dashboards.core/src/main/java/org/wso2/carbon/dashboards/core/internal/DashboardMetadataProviderImpl.java

```diff
@@ -164,15 +164,29 @@ public WidgetMetadataProvider getWidgetMetadataProvider() {
         Set<DashboardMetadata> dashboardList = dao.getAll();
         return dashboardList.stream().
                 filter(dashboardMetadata -> {
+                    DashboardMetadata dashboardMetadataDetails;
+                    try {
+                        Optional<DashboardMetadata> dashboardMetadataOptional = get(dashboardMetadata.getUrl());
+                        if (dashboardMetadataOptional.isPresent()) {
+                            dashboardMetadataDetails = dashboardMetadataOptional.get();
```

Comment:
what does this get? method name is a bit confusing.

Reply:
It is a method defined in Optional[1] used to get element of an optional result.

[1] https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html

Reply:
This is an already existing method[1]. Ideally it should be ```getDashboard()```

[1] https://github.com/wso2/carbon-dashboards/blob/d1c8fdd66182b45eacccb2510004275f8c61189e/components/dashboards/org.wso2.carbon.dashboards.core/src/main/java/org/wso2/carbon/dashboards/core/internal/DashboardMetadataProviderImpl.java#L143